src/components/multi-range/multi-range.component.ts

```diff
@@ -0,0 +1,380 @@
+ * @cssproperty --track-color-inactive - The of the portion of the track that represents the remaining value.
+ * @cssproperty --track-height - The height of the track.
+ */
+export default class SlMultiRange extends ShoelaceElement {
+  static styles: CSSResultGroup = [formControlStyles, styles];
+
+  /** The range's label. If you need to display HTML, use the `label` slot instead. */
+  @property() label = '';
+
+  /** The range's help text. If you need to display HTML, use the help-text slot instead. */
+  @property({ attribute: 'help-text' }) helpText = '';
+
+  /** Disables the range. */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  /** The minimum acceptable value of the range. */
+  @property({ type: Number }) min = 0;
+
+  /** The maximum acceptable value of the range. */
+  @property({ type: Number }) max = 100;
+
+  /** The interval at which the range will increase and decrease. */
+  @property({ type: Number }) step = 1;
+
+  /** The preferred placement of the range's tooltip. */
+  @property() tooltip: 'top' | 'bottom' | 'none' = 'top';
+
+  /** The current values of the range */
+  @property({ type: Array })
+  set value(value: readonly number[]) {
+    this.#value = value || [];
+  }
+  get value() {
+    return this.#value;
+  }
+
+  /**
+   * A function used to format the tooltip's value. The range's value is passed as the first and only argument. The
+   * function should return a string to display in the tooltip.
+   */
+  @property({ attribute: false }) tooltipFormatter: (value: number) => string = (value: number) => value.toString();
+
+  @query('.base') baseDiv: HTMLDivElement;
+  @query('.active-track') activeTrack: HTMLDivElement;
+  @query('.tooltip') tooltipElem: HTMLDivElement | undefined;
+  @queryAll('.handle') handles: NodeListOf<HTMLDivElement>;
+
+  #hasSlotController = new HasSlotController(this, 'help-text', 'label');
+  #value: readonly number[] = [0, 100];
+  #sliderValues = new Map<number, number>();
+  #hasFocus = false;
+  #nextId = 1;
+
+  override render(): unknown {
+    const hasLabel = !!(this.label || this.#hasSlotController.test('label'));
+    const hasHelpText = !!(this.helpText || this.#hasSlotController.test('help-text'));
+
+    const tooltip = this.tooltip !== 'none' ? html`<div class="tooltip" aria-hidden="true"></div>` : nothing;
+
+    this.#sliderValues.clear();
+    const handles = this.#value.map(value => {
+      const sliderId = this.#nextId++;
+      this.#sliderValues.set(sliderId, value);
+      return html`
+        <div
+          class="handle"
+          tabindex="${this.disabled ? -1 : 0}"
+          role="slider"
+          aria-labelledby=${ifDefined(hasLabel ? 'label' : undefined)}
+          aria-valuemin="${this.min}"
+          aria-valuemax="${this.max}"
+          aria-disabled=${ifDefined(this.disabled ? 'true' : undefined)}
+          aria-valuenow="${value}"
+          data-slider-id="${sliderId}"
+          @pointerdown=${this.#onClickHandle}
```

Comment:
I use the pointer* events instead of mouse events here because it gracefully and automatically handles edge cases like releasing the mouse button when the cursor is no longer in the browser window. It also means that you should  be able to move multiple sliders at once on multi-tap devices.